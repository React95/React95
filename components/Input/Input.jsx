import styled from 'styled-components';

const Input = styled.input`
  outline: none;
  border: none;

  padding: 3px 3px 5px 3px;

  border-left: 1px solid #868a8e;
  border-top: 1px solid #868a8e;

  box-shadow: inset -1px -1px 0 0 #c3c7cb, inset 1px 1px 0 0 #000000,
    0.5px 0.5px 0 0.5px #ffffff;
`;

export default Input;
