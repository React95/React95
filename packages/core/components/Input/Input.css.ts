import { style } from '@vanilla-extract/css';
import { Cursor } from '../Cursor/Cursor.css';
import { contract } from '../themes/contract.css';

export const input = style([
  Cursor.Text,
  {
    outline: 'none',
    border: 'none',
    paddingTop: contract.space[3],
    paddingRight: contract.space[3],
    paddingBottom: contract.space[5],
    paddingLeft: contract.space[3],
    color: contract.colors.materialText,
    backgroundColor: contract.colors.inputBackground,
    borderRadius: '0',
    borderTopWidth: contract.space[1],
    borderTopStyle: 'solid',
    borderTopColor: contract.colors.borderDark,
    borderRightWidth: '0',
    borderBottomWidth: '0',
    borderLeftWidth: contract.space[1],
    borderLeftStyle: 'solid',
    borderLeftColor: contract.colors.borderDark,
    boxShadow: contract.shadows.input,
    WebkitAppearance: 'none',
  },
]);
