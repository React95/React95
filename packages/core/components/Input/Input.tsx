import React, { forwardRef } from 'react';
import { Frame, FrameProps } from '../Frame/Frame';
import { input } from './Input.css';

export type InputProps = Omit<FrameProps<'input'>, 'as'>;

export const Input = forwardRef<HTMLInputElement, InputProps>((rest, ref) => (
  <Frame {...rest} ref={ref} className={input} as="input" />
));
