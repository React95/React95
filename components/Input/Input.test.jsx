import React from 'react';
import Input from './Input';
import { render } from 'react-testing-library';

describe('<Input />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Input />);
      expect(container).toMatchSnapshot();
    });
  });
});
