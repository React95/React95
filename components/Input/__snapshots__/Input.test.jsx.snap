// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshots should match snapshot 1`] = `
.c0 {
  outline: none;
  border: none;
  padding: 3px 3px 5px 3px;
  border-left: 1px solid #868a8e;
  border-top: 1px solid #868a8e;
  box-shadow: inset -1px -1px 0 0 #c3c7cb,inset 1px 1px 0 0 #000000,0.5px 0.5px 0 0.5px #ffffff;
}

<div>
  <input
    class="c0"
  />
</div>
`;
